soccer/vision/camera/World.cpp

```diff
@@ -0,0 +1,221 @@
+#include "World.hpp"
+
+#include <Constants.hpp>
+
+#include "vision/util/VisionFilterConfig.hpp"
+
+REGISTER_CONFIGURABLE(World)
+
+ConfigDouble* World::fast_kick_timeout;
+ConfigDouble* World::slow_kick_timeout;
+ConfigDouble* World::same_kick_timeout;
+
+void World::createConfiguration(Configuration* cfg) {
+    // Note: slow kick timeout should be smaller than fast kick timeout
+    fast_kick_timeout = new ConfigDouble(cfg, "VisionFilter/Kick/Detector/fast_kick_timeout", 1);
+    slow_kick_timeout = new ConfigDouble(cfg, "VisionFilter/Kick/Detector/slow_kick_timeout", 0.5);
+    same_kick_timeout = new ConfigDouble(cfg, "VisionFilter/Kick/Detector/same_kick_timeout", 0.05);
+}
+
+World::World()
+    : cameras(*VisionFilterConfig::max_num_cameras),
+      robotsYellow(Num_Shells, WorldRobot()),
+      robotsBlue(Num_Shells, WorldRobot()) {}
+
+void World::updateWithCameraFrame(RJ::Time calcTime, const std::vector<CameraFrame>& newFrames) {
+    calcBallBounce();
+
+    std::vector<bool> cameraUpdated(*VisionFilterConfig::max_num_cameras, false);
+
+    // TODO: Take only the newest frame if 2 come in for the same camera
```

Comment:
It should be about 60 hz. The filters are running at 100 hz.

I'm not sure I see how ignoring frames will lead to anomalies though. We would just be skipping the update phase so it'll never "fix" the estimation, but it shouldn't cause any major problems.

Additionally, if this happens where two frames do arrive within a small period in time due to network/SSL Vision lag on the older one, is it even a good idea to run the update step on both measurements at the current time since one will be at least 1/60th of a second out of date? There are filters that take that into account and do some cool stuff to account for that, but is it really worth it since the models themselves are not close to the physical processes they represent?

In the current state without removing old frames, it's possible that we may do 2 predicts and updates in a single frame which would cause some weird problems. I'll do a quick test tonight to see how often this actually happens.

(This isn't implemented right now, but I should probably be rejecting any out of order frames anyways)